src/main/java/frc/robot/subsystems/IndexerSubsystem.java

```diff
@@ -1,126 +1,133 @@
+  private final CANSparkMax rightIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_RIGHT_MOTOR, kBrushless);
+  private static SparkPIDController leftPidController;
+  private static SparkPIDController rightPidController;
 
-  public IndexerState indexerState;
+  private final ColorSensorV3 indexerColorSensor = new ColorSensorV3(Port.kOnboard);
+  private final ColorMatch indexerColorMatch = new ColorMatch();
 
-  public boolean capturedRing() {
-    return (this.indexerState != IndexerState.noRing);
-  }
 
-  public IndexerState getIndexerState() {
-    return indexerState;
-  }
+  public IndexerSubsystem() {
+    leftPidController = indexerMotorConfig(leftIndexerMotor, true);
+    rightPidController = indexerMotorConfig(rightIndexerMotor, false);
 
-  private boolean isIntakeSensorTripped() {
-    return colorSensorReader.getIntakeValues().proximity > THRESHOLD_INTAKE;
+    indexerColorMatch.addColorMatch(COLOR_NOTE);
+    indexerColorMatch.addColorMatch(COLOR_NONE);
   }
-
-  public boolean isShooterSensorTripped() {
-    return colorSensorReader.getShooterValues().proximity > THRESHOLD_SHOOTER;
+  
+  public boolean shouldContinue() {
+    return indexerColorMatch.matchClosestColor(indexerColorSensor.getColor()).color == COLOR_NOTE;
   }
 
-  public void stop() {
-    pidController.setReference(0, ControlType.kVelocity);
+  // Check actual motor values for dash board debugging
+  public boolean isActive() {
+    return leftIndexerMotor.get() > 0 || rightIndexerMotor.get() > 0;
   }
 
-  public void load() {
-    pidController.setReference(IndexerConstants.BELT_RUN_SPEED, ControlType.kVelocity);
+  public void runIndexer() {
+    leftPidController.setReference(BELT_RUN_SPEED, ControlType.kVelocity);
+    rightPidController.setReference(BELT_RUN_SPEED, ControlType.kVelocity);
   }
-  private final CANSparkMax leftIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_LEFT_MOTOR, kBrushless);
-  private final CANSparkMax rightIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_RIGHT_MOTOR, kBrushless);
-  private static SparkPIDController pidController;
-  private final ColorSensorReader colorSensorReader = new ColorSensorReader();
-  private final Notifier colorSensorNotifier = new Notifier(colorSensorReader);
-
-  private SysIdRoutine indexerSysIdRoutine = new SysIdRoutine(
-    new SysIdRoutine.Config(null, null, null, SysIdRoutineSignalLogger.logState()),
-    new SysIdRoutine.Mechanism((volts) -> {
-      leftIndexerMotor.setVoltage(volts.in(Volts));
-      rightIndexerMotor.setVoltage(volts.in(Volts));
-    }, null, this));
-
-  public IndexerSubsystem() {
-    indexerMotorConfig(leftIndexerMotor, true);
-    indexerMotorConfig(rightIndexerMotor, false);
 
-    colorSensorReader.run();
-    // Update the color sensors in the background to prevent loop overrun
-    colorSensorNotifier.setName("Color Sensors");
-    colorSensorNotifier.startPeriodic(0.02);
+  public void stopIndexer() {
+    leftPidController.setReference(0, ControlType.kVelocity);
```

Comment:
I'm not sure we want to set the velocity to 0, vs. setting the output to 0. If the motors are running, this will apply negative output to stop the motors. This will also apply current to hold the velocity at zero if the motors try to spin. Although we _might_ need this to hold the note in, I'm pretty sure the compression will keep it secure without running the motors.